File: src/test/resources/projects/testJavaProject/reference/src/test/java/com/company/bonitasoft/ConnectorTest.java
package com.company.bonitasoft;

import java.util.HashMap;
import java.util.Map;

import org.bonitasoft.engine.connector.ConnectorValidationException;
import org.junit.Test;

public class ConnectorTest {

    @Test(expected = ConnectorValidationException.class)
    public void should_throw_exception_if_mandatory_input_is_missing() throws ConnectorValidationException {
        Connector connector = new Connector();
        connector.validateInputParameters();
    }

    @Test(expected = ConnectorValidationException.class)
    public void should_throw_exception_if_mandatory_input_is_empty() throws ConnectorValidationException {
        Connector connector = new Connector();
        Map<String, Object> parameters = new HashMap<>();
        parameters.put(Connector.DEFAULT_INPUT, "");
        connector.setInputParameters(parameters);
        connector.validateInputParameters();
    }

    @Test(expected = ConnectorValidationException.class)
    public void should_throw_exception_if_mandatory_input_is_not_a_string() throws ConnectorValidationException {
        Connector connector = new Connector();
        Map<String, Object> parameters = new HashMap<>();
        parameters.put(Connector.DEFAULT_INPUT, 38);
        connector.setInputParameters(parameters);
        connector.validateInputParameters();
    }

    @Test
    public void should_accept_valid_input() throws ConnectorValidationException {
        Connector connector = new Connector();
        Map<String, Object> parameters = new HashMap<>();
        parameters.put(Connector.DEFAULT_INPUT, "valid");
        connector.setInputParameters(parameters);
        connector.validateInputParameters();
    }

}

File: src/test/resources/projects/testKotlinProject/reference/src/test/kotlin/com/company/bonitasoft/ConnectorTest.kt
package com.company.bonitasoft

import org.junit.Test
import org.bonitasoft.engine.connector.ConnectorValidationException

class ConnectorTest {

    @Test(expected = ConnectorValidationException::class)
    fun should_throw_exception_if_mandatory_input_is_missing() {
        val connector = Connector()
        connector.validateInputParameters()
    }
    
    @Test(expected = ConnectorValidationException::class)
    fun should_throw_exception_if_mandatory_input_is_empty() {
        val connector = Connector()
        connector.setInputParameters(mapOf("defaultInput" to ""))
        connector.validateInputParameters()
    }
    
    @Test(expected = ConnectorValidationException::class)
    fun should_throw_exception_if_mandatory_input_is_not_a_string() {
        val connector = Connector()
        connector.setInputParameters(mapOf("defaultInput" to 38))
        connector.validateInputParameters()
    }
    
    @Test
    fun should_accept_valid_input() {
        val connector = Connector()
        connector.setInputParameters(mapOf("defaultInput" to "valid"))
        connector.validateInputParameters()
    }
}